lib/pengine/utils.c

```diff
@@ -541,6 +541,20 @@ custom_action(pe_resource_t * rsc, char *key, const char *task,
                                        FALSE, data_set);
         }
 
+        // Make the action optional if its resource is unmanaged
+        if (!pcmk_is_set(action->flags, pe_action_pseudo)
+            && (action->node != NULL)
```

Comment:
> this meant that if a resource is unmanaged *and* its allocated node is pending, any probe would be set to optional, but not unrunnable

Are we now guaranteed to hit the "clear runnable" block in L580-585 for probes on pending nodes?

If so, looks good to me.

Reply:
Yes. The issue was that the test for an unmanaged resource (making the action optional) was previously an earlier link in that if-else chain, so if that was hit, it would never proceed to the is-node-pending test.

Now, the unmanaged test is in its own if block, without an else, so the runnable tests are always checked separately. Probes aren't pseudo-actions, so they won't hit the first test. If the node is pending, it's not null or offline, so it won't hit those tests either, but that wouldn't matter because it would still end up unrunnable anyway.